ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -548,20 +557,35 @@ private IndexValue findKey(StoreKey key, FileSpan fileSpan, IndexEntryType type,
           Iterator<IndexValue> it = values.descendingIterator();
           while (it.hasNext()) {
             IndexValue value = it.next();
+            if (latest == null) {
+              latest = value;
+            }
             logger.trace("Index : {} found value offset {} size {} ttl {}", dataDir, value.getOffset(), value.getSize(),
                 value.getExpiresAtMs());
-            if (type.equals(IndexEntryType.ANY)) {
-              retValue = value;
+            if (types.contains(IndexEntryType.DELETE) && value.isFlagSet(IndexValue.Flags.Delete_Index)) {
+              retCandidate = value;
               break;
-            } else if (type.equals(IndexEntryType.DELETE) && value.isFlagSet(IndexValue.Flags.Delete_Index)) {
-              retValue = value;
+            } else if (types.contains(IndexEntryType.TTL_UPDATE) && !value.isFlagSet(IndexValue.Flags.Delete_Index)
```

Comment:
I don't see any non-test code interested in TTL_UPDATE. `findKey()` calls everywhere seem to be interested only in `PUT` or `DELETE`. Are you going to include `TTL_UPDATE` in a subsequent patch?

Reply:
hmm.. I thought I added a test for this. Will recheck (maybe did it in one of the branches I have on top of these changes)

Reply:
Yeah, my comment was that there doesn't seem to be any 'non-test' code passing in TTL_UPDATE...

Reply:
oh missed the "non_test" part ! Yes, nothing in this PR accesses it. The `BlobStore` and `BlobStoreCompactor` will use it.